### PR TITLE
Expansion

### PR DESCRIPTION
**Pull request by buchi**

### CODE REVIEW

File: docs/source/_json/collection.json [hunk 1]
```diff
@@ -5,8 +5,19 @@ HTTP 200 OK
 content-type: application/json
 
 {
+  "@components": {
```
Comment (buchi) [inline docs/source/_json/collection.json]: @tisto regarding nesting I'm fine with both options.

Expansion works on "elements". Thus you can expand breadcrumbs and navigation separately. "Components" is not an expandable element.

File: docs/source/_json/collection.json [hunk 2]
```diff
@@ -5,8 +5,19 @@ HTTP 200 OK
...
+    "breadcrumbs": {
+      "@id": "http://localhost:55001/plone/collection/@components/breadcrumbs"
+    }, 
+    "navigation": {
+      "@id": "http://localhost:55001/plone/collection/@components/navigation"
+    }
+  }, 
   "@id": "http://localhost:55001/plone/collection", 
   "@type": "Collection", 
+  "@workflow": {
```
Comment (buchi) [inline docs/source/_json/collection.json]: @tisto Right now I'm using the "@"-prefix mainly for namespacing reasons. But that problem is not fully solved yet. We need a solution for handling name conflicts with field names if fields are on top level of the JSON representation.

**Comment by tisto** [comment]
@buchi @bloodbare @lukasgraf @sneridagh As discussed at the Beethoven Sprint I added the docs for expansion. I hope I got it right from my notes, sorry it took me so long to write that down. Please review my documentation and let me know if this is what we agreed upon:

https://github.com/plone/plone.restapi/blob/81e600ec65a5e926a8e382fc3750f9ffa45b7ad3/docs/source/expansion.rst

@buchi I did not touch any code. I hope it is ok that I high-jacked your branch.

**Comment by buchi** [comment]
@tisto Thanks for adding the docs. Im going to update the implementation...

**Comment by tisto** [comment]
I started to look into amending the implementation. Right now expansion only works for breadcrumbs/navigation with the  "@components/" prefix.

I already added breadcrumbs/navigation as top-level components:

https://github.com/plone/plone.restapi/pull/312

The next step would be to amend the expansion adapter implementation to support all our top-level components (breadcrumbs, navigation, schema, workflow, etc.).

@buchi @lukasgraf @bloodbare @sneridagh could you please confirm that we decided to go with top-level components instead of using the "@components"-prefix?

Please check my documentation to make sure we are all on the same page before proceeding:

https://github.com/plone/plone.restapi/blob/a0dbb5f9ceeafbf96aa8788de1899d6d1ef81520/docs/source/expansion.rst

I would really love to finish this in the next weeks because this is the one major blocker for a final release.

**Comment by buchi** [comment]
@tisto correct. The docs represent what we agreed to.

I'm going to continue here. Bah! Rebase hell...

**Comment by tisto** [comment]
@buchi awesome! Let me know if there is anything I can do to help! I am online on the plone slack channel. Feel free to ping me...

**Comment by buchi** [comment]
@tisto What's the reason you commented out the new `@breadcrumbs` and `@navigation` endpoints? We need `@components` now for expansion. Thus I would like to have breadcrumbs and navigation toplevel.

The generated samples now contain updated responses. Please check if you agree.
https://github.com/plone/plone.restapi/blob/expansion/docs/source/_json/expansion.resp
https://github.com/plone/plone.restapi/blob/expansion/docs/source/_json/expansion_expanded.resp
https://github.com/plone/plone.restapi/blob/expansion/docs/source/_json/expansion_expanded_full.resp

What else should be expandable?  We don't have a `@schema` endpoint yet. There's `@types` but that doesn't fit for content.

BTW how can I join the plone channel on slack?

**Comment by ebrehault** [comment]
> BTW how can I join the plone channel on slack?

@buchi just register here: http://slackin-toplone.rhcloud.com/

**Comment by tisto** [comment]
@buchi commenting out navigation and breadcrumbs was just a temporary solution to get a release out. I did not want to release the new endpoints without your approval. I saw that you re-added them. Will look into the pr now.

I was under the assumption that slack would be somehow synced with IRC/gitter. I will be online on good old IRC as well during the day. :)

**Comment by tisto** [comment]
@buchi LGTM! Can't wait to push the merge button! :)

I will write a few more tests and play around with it...

**Comment by ebrehault** [comment]
merge it! merge it! merge it! merge it!

**Comment by sneridagh** [comment]
Woohooo!

**Pull request merged**